h2o-core/src/main/java/hex/Model.java

```diff
@@ -1156,7 +1156,6 @@ else if (isWeights && computeMetrics) {
           if (expensive) {
             vec = test.anyVec().makeCon(1);
             toDelete.put(vec._key, "adapted missing vectors");
-            msgs.add(H2O.technote(1, "Test/Validation dataset is missing weights column '" + names[i] + "' (needed because a response was found and metrics are to be computed): substituting in a column of 1s"));
```

Comment:
This warning should not be removed, we need to make sure it is removed just for the right cases.

For pure predict, we have a PR that addresses the issue of redundant warnings: https://github.com/h2oai/h2o-3/pull/2263